Mathlib/Data/Finset/Density.lean

```diff
@@ -0,0 +1,217 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Algebra.Order.Field.Basic
+import Mathlib.Tactic.Positivity.Basic
+import Mathlib.Tactic.Positivity.Finset
+
+/-!
+# Density of a finite set
+
+This defines the density of a `Finset` and provides induction principles for finsets.
+
+## Main declarations
+
+* `Finset.dens t`: `dens s : ℕ` returns the density of `s : Finset α`.
+
+## Notation
+
+If you need to specify the field the density is valued in, `dens[𝕜] s` is notation for
+`(dens s : 𝕜)`. Note that no dot notation is allowed.
+-/
+
+open Function Multiset Nat
+
+variable {𝕜 α β : Type*} [Fintype α]
+
+namespace Finset
+section Semifield
+variable [Semifield 𝕜] {s t : Finset α} {a b : α}
+
+/-- `dens s` is the number of elements of `s`, aka its density. -/
+@[pp_dot] def dens (s : Finset α) : 𝕜 := s.card / Fintype.card α
+
+@[inherit_doc dens] notation "dens[" 𝕜 "]" => @dens 𝕜
+
+lemma card_div_card_eq_dens (s : Finset α) : dens[𝕜] s = s.card / Fintype.card α := rfl
+
+@[simp] lemma dens_empty : dens[𝕜] (∅ : Finset α) = 0 := by simp [dens]
+
+@[simp] lemma dens_singleton (a : α) : dens ({a} : Finset α) = (Fintype.card α : 𝕜)⁻¹ := by
+  simp [dens]
+
+@[simp]
+lemma dens_cons (h : a ∉ s) : (s.cons a h).dens = dens s + (Fintype.card α : 𝕜)⁻¹ := by
+  simp [dens, add_div]
+
+@[simp]
+lemma dens_disjUnion (s t : Finset α) (h) : dens[𝕜] (s.disjUnion t h) = dens s + dens t := by
+  simp_rw [dens, card_disjUnion, Nat.cast_add, add_div]
+
+section Lattice
+variable [DecidableEq α]
+
+lemma dens_union_add_dens_inter (s t : Finset α) :
+    dens[𝕜] (s ∪ t) + dens (s ∩ t) = dens s + dens t := by
+  simp_rw [dens, ← add_div, ← Nat.cast_add, card_union_add_card_inter]
+
+lemma dens_inter_add_dens_union (s t : Finset α) :
+    dens[𝕜] (s ∩ t) + dens (s ∪ t) = dens s + dens t := by rw [add_comm, dens_union_add_dens_inter]
+
+@[simp] lemma dens_union_of_disjoint (h : Disjoint s t) : dens[𝕜] (s ∪ t) = dens s + dens t := by
+  rw [← disjUnion_eq_union s t h, dens_disjUnion _ _ _]
+
+lemma dens_sdiff_add_dens_eq_dens (h : s ⊆ t) :  dens[𝕜] (t \ s) + dens s = dens t := by
```

Comment:
```suggestion
lemma dens_sdiff_add_dens_eq_dens (h : s ⊆ t) : dens[𝕜] (t \ s) + dens s = dens t := by
```

Reply:
This matches `card` lemmas, so if you want that name to change you will first need to rename the corresponding `card` lemma